integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/BookResource.java

```diff
@@ -58,4 +59,33 @@ public Response findByPublicationYear(@PathParam("year") Integer year) {
         Optional<Book> book = bookRepository.findByPublicationYear(year);
         return book.map(b -> Response.ok(book).build()).orElse(Response.noContent().build());
     }
+
+    @GET
+    @Path("/count/year")
+    @Produces("application/json")
+    public List<BookRepository.BookCountByYear> countAllByPublicationYear() {
```

Comment:
I am also thinking that the tests should include another custom type as well, just to be sure nothing weird gets tripped

Reply:
Will add another. Give me a few. Will use at least one different return type. I'm afraid my brain fails at anything useful for a Join query.

Reply:
:+1: